src/irmin-pack/inode_intf.ml

```diff
@@ -61,12 +61,18 @@ module type Internal = sig
 
   exception Dangling_hash of { context : string; hash : hash }
 
-  module Raw : Pack_value.S with type hash = hash
+  module Raw : sig
+    include Pack_value.S with type hash = hash
+
+    val depth : t -> int option
+
+    exception Invalid_depth of { expected : int; got : int option; v : t }
```

Comment:
we can maybe add a test to check that this is indeed raised as expected? (we can do that in a follow up issue too)

Reply:
Tracked in #1668